### PR TITLE
feat: introduce MapStreamFn for streaming map udf

### PR DESCRIPTION
**Pull request by xdevxy**

### CODE REVIEW

File: src/main/java/io/numaproj/numaflow/function/handlers/MapStreamHandler.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import io.grpc.stub.StreamObserver;
+import io.numaproj.numaflow.function.interfaces.Datum;
+import io.numaproj.numaflow.function.types.Message;
+import io.numaproj.numaflow.function.v1.Udfunction.DatumResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * MapHandler exposes method for performing map streaming operation.
```
Comment (KeranYang) [inline src/main/java/io/numaproj/numaflow/function/handlers/MapStreamHandler.java:13]:
```suggestion
 * MapStreamHandler exposes method for performing map streaming operation.
```

File: src/main/java/io/numaproj/numaflow/function/FunctionServer.java [hunk 2]
```diff
@@ -90,6 +91,16 @@ public FunctionServer registerMapTHandler(MapTHandler mapTHandler) {
         return this;
     }
 
+    /**
+     * registers the map stream handler to the server.
+     * @param mapStreamHandler handler to process the message and assign event time.
+     * @return returns a new Function gRPC server.
+     */
+    public FunctionServer registerMapStreamHandler(MapStreamHandler mapStreamHandler) {
```
Comment (KeranYang) [inline src/main/java/io/numaproj/numaflow/function/FunctionServer.java:99]: Could you add a unit test at FunctionServerTest to cover map stream? How was this PR tested?

**Review by KeranYang — CHANGES_REQUESTED** [review]

File: src/main/java/io/numaproj/numaflow/function/FunctionServer.java (hunk 2 above)
Comment (xdevxy) [reply src/main/java/io/numaproj/numaflow/function/FunctionServer.java:99]: Had it in the local branch missed it to commit...

Comment (KeranYang) [reply src/main/java/io/numaproj/numaflow/function/FunctionServer.java:99]: thanks! My second question is about sdk test, do we plan to add it for Java?

**Review by KeranYang — APPROVED** [review]

File: src/main/java/io/numaproj/numaflow/function/FunctionServer.java (hunk 2 above)
Comment (xdevxy) [reply src/main/java/io/numaproj/numaflow/function/FunctionServer.java:99]: yes, going to update the e2e test with java and python.

**Pull request merged**